testbed/tests/widgets/test_optioncontainer.py

```diff
@@ -90,6 +90,47 @@ def assert_tab_text(tab, expected):
     assert type(tab.text) is str
 
 
+async def test_content_size_rehint(
+    widget, probe, content1, content1_probe, content3, main_window, main_window_probe
+):
+    """The OptionContainer should rehint to minimum size of all widgets plus necessary
+    decors, and should automatically rehint upon the addition of new tabs."""
+    skip_on_platforms(
+        "iOS",
+        "android",
+        "windows",
+        reason="Accurate OptionContainer hinting is not yet"
+        "implemented on this platform",
+    )
+    old_main_window_size = main_window.size
+    main_window.size = (300, 300)
+    await main_window_probe.redraw("Main window size should be 300x300")
+
+    content3.width = 500
+    content3.height = 600
+    # 1 seconds to allow events to propagate properly, since the refreshment happens
+    # across multiple event loop iterations.
+    # 1 second is chosen because some CI machines can be notoriously slow with our
+    # GTK implementation.
+    await probe.redraw("Tab 3's size should be explicitly set to 500x600", delay=1)
+
+    try:
+        assert main_window.size.width >= 500
+        assert main_window.size.height > 600
+        assert probe.width >= 500
+        assert probe.height > 600
+        # Asserting content1 for content size because content3, by virtue of not being
+        # selected, may have an invalid size.
+        assert content1_probe.width == 500
+        assert content1_probe.height == 600
+    finally:
+        del content3.width
+        del content3.height
+        await probe.redraw("Cleanup: Removing width and height requirements", delay=1)
+        main_window.size = old_main_window_size
+        await probe.redraw("Cleanup: Resizing window")
```

Comment:
Why is this cleanup needed? The widget should be created fresh for each widget; if the issue is window size being altered, then that cleanup should be part of the Window fixture so it is inherited everywhere.